### DISCUSSION TITLE
A possible solution to reliably exit marks?

### DISCUSSION THREAD

**Discussion by cadars** [ueberdosis/tiptap#2160] [Feature Requests]
Someone wrote an interesting prosemirror plugin for `code` marks, involving a fake cursor: https://discuss.prosemirror.net/t/new-plugin-prosemirror-codemark-to-handle-inline-code-mark-creation-and-manipulation/4159

Related: https://github.com/ueberdosis/tiptap/pull/1200

**Comment by rowanc1** [comment]
Maintainer of that `prosemirror-codemark` here 👋. Happy to make changes in the underlying lib as well there it there are changes necessary to be consumed by tiptap!

**Reply by cadars** [reply]
Thanks for this and for chiming in! I just wanted to bring your elegant solution to the attention of the tiptap community, after spending a lot of time fiddling with the somehow too drastic `inclusive: false / true` setting in tiptap.

**Reply by rowanc1** [reply]
Yeah - code marks are a feature that are done poorly in a lot of wysiwyg editors, and I think in some ways this needs to be `inclusive` in the prosemirror setting, but give you a way to arrow out on both sides. The feature is surprisingly complex to get it right, the lib still needs some work for some of the browsers, but it works a lot better than the default. :)

**Reply by hanspagel** [reply]
Thanks for bringing this up @cadars! We’re closely following all ProseMirror-related things and have seen this already. Amazing work @rowanc1! 🤙

We don’t have an opinion whether we’ll integrate it yet, but definitely an interesting approach.

**Reply by rowanc1** [reply]
Let me know if you do go in this direction!

**Reply by nokola** [reply]
I think this plugin is awesome! I managed to use it in TipTap, with workaround. Opened a bug for the TipTap issue, which I think would be worth fixing since it will help other similar plugins (that need schema before editor is created.) #2227

**Reply by rowanc1** [reply]
Nice. Looks like it was `this.editor.schema` for the fix, glad you got it working! :) Anything to add to `prosemirror-codemark`'s documentation? Maybe a section on "Using with TipTap" or something?

**Reply by nokola** [reply]
Added in https://github.com/curvenote/prosemirror-codemark/pull/6 which also adds a small dev quality-of-life improvement for local debugging.

**Comment by lukesmurray** [comment]
This issue also shows up for any custom inline elements. For example, given the following content, you would not be able to add text at the end of the paragraph. Any text would be extending the `inline-element`.

`<p>Hello World <inline-element>inline text content</inline-element></p>`

![Kapture 2021-12-10 at 14 43 58](https://user-images.githubusercontent.com/34020210/145632513-e55d085d-a7a4-4acd-94df-7cd79f0728d7.gif)

I feel like the expected UX would be pressing right arrow exits the inline node but the selection stays in the same block, pressing enter adds a new block below the inline node. Not sure if there is a reasonable way to implement this behavior right now since I just started to check out tiptap but documentation for accomplishing those two tasks would be helpful!
